File: examples/Clojure/psenvpub.clj
(ns psenvpub
  (:refer-clojure :exclude [send])
  (:require [zhelpers :as mq]))

;;
;; Pubsub envelope publisher
;; Isaiah Peng <issaria@gmail.com>
;;

(defn -main []
  (let [ctx (mq/context 1)
        publisher (mq/socket ctx mq/pub)]
    (mq/bind publisher "tcp://*:5563")
    (while true
      ;; Write two messages, each with an envelope and content
      (mq/send publisher "A\u0000" mq/sndmore)
      (mq/send publisher "We don't want to see this.\u0000")
      (mq/send publisher "B\u0000" mq/sndmore)
      (mq/send publisher "We would like to see this.\u0000")
      (Thread/sleep 1000))
    (.close publisher)
    (.term ctx)))

File: examples/Go/psenvpub.go
//
//  Pubsub envelope publisher
//

package main

import (
	zmq "github.com/alecthomas/gozmq"
	"time"
)

func main() {
	context, _ := zmq.NewContext()
	defer context.Close()

	publisher, _ := context.NewSocket(zmq.PUB)
	defer publisher.Close()
	publisher.Bind("tcp://*:5563")

	for {
		publisher.SendMultipart([][]byte{[]byte("A"), []byte("We don't want to see this")}, 0)
		publisher.SendMultipart([][]byte{[]byte("B"), []byte("We would like to see this")}, 0)
		time.Sleep(time.Second)
	}
}
